### PR TITLE
Fix daedalean-include-order to work with the new CPP layout

### PR DESCRIPTION
**Pull request by Javier-varez**

### CODE REVIEW

File: clang-tools-extra/clang-tidy/daedalean/IncludeOrderCheck.cpp [hunk 1]
```diff
@@ -11,6 +11,11 @@
 #include "clang/ASTMatchers/ASTMatchFinder.h"
 #include "clang/Lex/PPCallbacks.h"
 #include "clang/Lex/Preprocessor.h"
+#include "llvm/ADT/None.h"
+#include "llvm/ADT/Optional.h"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "daedalean-include-order"
```
Comment (finomen) [inline clang-tools-extra/clang-tidy/daedalean/IncludeOrderCheck.cpp:18]: What is this define doing?

Comment (Javier-varez) [reply clang-tools-extra/clang-tidy/daedalean/IncludeOrderCheck.cpp:18]: It is needed for `LLVM_DEBUG`, to filter debug messages with `-debug-only=daedalean-include-order`

Comment (Javier-varez) [reply clang-tools-extra/clang-tidy/daedalean/IncludeOrderCheck.cpp:18]: https://llvm.org/docs/ProgrammersManual.html#the-llvm-debug-macro-and-debug-option

**Comment by Javier-varez** [comment]
@finomen, could you have a look at this when you get some time? thanks!

**Review by finomen — APPROVED** [review]

**Pull request merged**